### PR TITLE
Add external-dns to EKS cluster

### PR DESCRIPTION
**Pull request by giorgiosironi**
For https://github.com/elifesciences/issues/issues/4572, adds a `external-dns` Deployment to a EKS cluster that synchronizes Route53 DNS for `elifesciences.org` with that is required by `LoadBalancer`-type `Service` annotations.

The resulting capability is that when creating a release of an Helm chart, a new `CNAME` (actually a `A` alias) get created pointing to the `Service`; and removed when the release is deleted.

## Tasks
- [x] Add a generated `AmazonRoute53KubernetesExternalDNS` policy
- [x] Attach that policy to `kubernetes-aws--test--AmazonEKSWorkerRole`
- [ ] Add a `helm_release` of `external-dns` chart

### CODE REVIEW

File: .activate-venv.sh [hunk 1]
```diff
@@ -17,10 +17,10 @@ fi
 # prefer python2 over python3
 if [ ! -f .use-python-3.flag ]; then
     # highest installed version of py2
-    python=$(which python2)
+    python=$(which python2 python2.7 | head -n 1)
 else
     # python 3.5
-    python=/usr/bin/python3.5
+    python=$(which python3.5 python3 | head -n 1)
```
Comment (giorgiosironi) [inline .activate-venv.sh]: change order of `python3*` for consistency with `python2*` cases

File: projects/elife.yaml [hunk 2]
```diff
@@ -2166,6 +2167,8 @@ kubernetes-aws:
                 max-size: 4
                 desired-capacity: 2
             helm: true
+            external-dns:
+                domain-filters: "elifesciences.net"
```
Comment (giorgiosironi) [inline projects/elife.yaml]: should be singular

File: src/buildercore/terraform.py [hunk 3]
```diff
@@ -890,6 +928,11 @@ def _render_helm(context, template):
         ],
     })
 
+    #template.populate_data(DATA_TYPE_HELM_REPOSITORY, DATA_NAME_HELM_STABLE, block={
+    #    'name': 'stable',
+    #    'url': 'https://kubernetes-charts.storage.googleapis.com',
+    #})
```
Comment (giorgiosironi) [inline src/buildercore/terraform.py]: remove as not needed?

File: src/buildercore/terraform.py [hunk 4]
```diff
@@ -903,6 +946,49 @@ def _render_helm(context, template):
         'depends_on': ['kubernetes_cluster_role_binding.tiller'],
     })
 
+    if context['eks']['external-dns']:
+        template.populate_resource('helm_release', 'external_dns', block={
+            'name': 'external-dns',
+            #'repository': "${data.helm_repository.%s.metadata.0.name}" % DATA_NAME_HELM_INCUBATOR,
```
Comment (giorgiosironi) [inline src/buildercore/terraform.py]: remove if not needed

File: src/buildercore/terraform.py [hunk 5]
```diff
@@ -903,6 +946,49 @@ def _render_helm(context, template):
...
+            'chart': 'stable/external-dns',
+            'version': HELM_CHART_VERSION_EXTERNAL_DNS,
+            'depends_on': ['helm_release.common_resources'],
+    # set {
+    #    name  = "cluster.enabled"
+    #    value = "true"
+    #  }
```
Comment (giorgiosironi) [inline src/buildercore/terraform.py]: remove, it's an example?

File: .activate-venv.sh [hunk 6]
```diff
@@ -17,10 +17,10 @@ fi
 # prefer python2 over python3
 if [ ! -f .use-python-3.flag ]; then
     # highest installed version of py2
-    python=$(which python2)
+    python=$(which python2 python2.7 | head -n 1)
```
Comment (lsh-0) [inline .activate-venv.sh]: this is clever. see https://github.com/elifesciences/lax/blob/develop/mkvenv.sh for something similar

File: .activate-venv.sh [hunk 7]
```diff
@@ -32,15 +32,14 @@ if [ ! -e "venv/bin/$py" ]; then
...
+if [ ! -f .use-python-3.flag ]; then
+    virtualenv --python=$python venv
+else
+    python3 -m venv venv
+fi
 source venv/bin/activate
 
-if [ "$(uname)" = "Darwin" ]; then
-    # 'ARCHFLAGS' fixes a problem with OSX refusing to compile a dependency
-    export ARCHFLAGS="-Wno-error=unused-command-line-argument-hard-error-in-future"
```
Comment (lsh-0) [inline .activate-venv.sh]: good riddance

File: projects/elife.yaml [hunk 8]
```diff
@@ -145,6 +145,9 @@ defaults:
                 # (currently fixed) number of EC2 nodes
                 desired-capacity: 1
             helm: false
+            external-dns: false
+                # Route53 hosted zone to create subdomains on
+                #domain-filter: elifesciences.org
```
Comment (lsh-0) [inline projects/elife.yaml]: is `domain-filter` part of the EKS vocabulary?

Comment (giorgiosironi) [reply projects/elife.yaml]: it comes from https://github.com/helm/charts/tree/master/stable/external-dns#configuration but in the singular form here; more `external-dns` vocabulary as you could use it with GCP and its Cloud DNS for example

**Pull request merged**